assets/js/modules/analytics/components/common/PropertySelectIncludingGA4.test.js

```diff
@@ -0,0 +1,174 @@
+import { MODULES_ANALYTICS, ACCOUNT_CREATE } from '../../datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import * as fixtures from '../../datastore/__fixtures__';
+import * as analytics4Fixtures from '../../../analytics-4/datastore/__fixtures__';
+
+import { fireEvent, act, render } from '../../../../../../tests/js/test-utils';
+
+const {
+	createProperty,
+	createWebDataStream,
+	properties: propertiesGA4,
+	webDataStreams,
+} = analytics4Fixtures;
+const { accounts, properties: propertiesUA, profiles } = fixtures.accountsPropertiesProfiles;
+const accountID = createProperty._accountID;
+const propertyIDga4 = createWebDataStream._propertyID;
+const propertyIDua = propertiesUA[ 0 ].id;
+
+const setupRegistry = ( { dispatch } ) => {
+	dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: 'http://example.com' } );
+	dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+	dispatch( MODULES_ANALYTICS ).receiveGetExistingTag( null );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetProperties( propertiesUA, { accountID } );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getProperties', [ accountID ] );
+
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetProperties( propertiesGA4, { accountID } );
+	dispatch( MODULES_ANALYTICS_4 ).finishResolution( 'getProperties', [ accountID ] );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetProfiles( profiles, { accountID, propertyID: propertyIDua } );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getProfiles', [ accountID, propertyIDua ] );
+
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetWebDataStreams( webDataStreams, { propertyID: propertyIDga4 } );
+	dispatch( MODULES_ANALYTICS_4 ).finishResolution( 'receiveGetWebDataStreams', { propertyID: propertyIDga4 } );
+};
+
+const setupEmptyRegistry = ( { dispatch } ) => {
+	dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetProperties( [], { accountID } );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getProperties', [ accountID ] );
+
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetProperties( [], { accountID } );
+	dispatch( MODULES_ANALYTICS_4 ).finishResolution( 'getProperties', [ accountID ] );
+};
+
+describe( 'PropertySelectIncludingGA4', () => {
+	it( 'should render an option for each analytics property of the currently selected account.', async () => {
```

Comment:
`async` isn't needed here because we don't wait for anything in the test. Please, remove it from all tests that don't use async/await functionality.

Reply:
Done for all the new tests I've added @eugene-manuilov . Nice spot! :smile: